### PR TITLE
[1156] Expose a Veda UI EnvConfigProvider for providing env variables

### PR DESCRIPTION
**Pull request by dzole0311**
**Related Ticket:** https://github.com/NASA-IMPACT/veda-ui/issues/1166

### Problem

We need a way for future Next.js instances using veda-ui components to pass their environment variables to the library

[The previous PR](https://github.com/NASA-IMPACT/veda-ui/pull/1252) tried to pass env variables as props to our components. However, this might be difficult to maintain and to scale as new environment variables will be added.

This PR introduces a centralized config place through a new `VedauiConfigProvider` context to allow devs to configure all environment variables in one place via a config object.

### Description of Changes

**Veda-UI changes**

- Created `VedauiConfigProvider` context provider
- Currently supports these configuration options:
  - `mapboxToken` ( `MAPBOX_TOKEN`)
  - `apiRasterEndpoint` (`API_RASTER_ENDPOINT`)
  - `apiStacEndpoint` (from `API_STAC_ENDPOINT`)
- Refactored internal components to consume the environment variables from this context

**Next.js ([PR here](https://github.com/developmentseed/next-veda-ui/pull/13)):**

- Imported and used the new provider
- ℹ️ In Next.js, environment variables should be named according to the Next.js naming standards so that they can be applied e.g `MAPBOX_TOKEN` should be `NEXT_PUBLIC_MAPBOX_TOKEN` etc. 

### Notes & Questions About Changes
_{Add additonal notes and outstanding questions here related to changes in this pull request}_

### Validation / Testing

Test that the correct env variables are applied when using the library in Next.js

### CODE REVIEW

**Comment by hanbyul-here** [comment]
Thanks for showing both approaches @dzole0311 🙇 

Yah I agree with @sandrahoang686 . The need for something that can save the configuration-ish (I am saying -ish since I don't think all the configurations can go to providers.) data has been rising, and this might be time to do so. 

This is a bit of implementation detail that doesn't have to be tackled now, but mainly to jot down my thoughts -
1.  The other properties that I think worth going to Context is path for each page ex. e&a page needs dataset page route and vice versa - and I think it is confusing for users to pass datasetpath as a prop to e&a page. (and of course, link) 
2. I wonder if it is better to differentiate the values that come from dotenv vs. not -ex. the context  assigns the initial value from dotenv without asking the client to hydrate.

**Comment by dzole0311** [comment]
@hanbyul-here @sandrahoang686 thanks for the comments!

> The other properties that I think worth going to Context is path for each page ex. e&a page needs dataset page route and vice versa - and I think it is confusing for users to pass datasetpath as a prop to e&a page. (and of course, link)

@hanbyul-here Agreed, this would definitely be confusing. But I think we should provide all route-triggering actions as callbacks within the components so that components don't have to handle routing directly (if that's feasible). 

> (and of course, link) 

For the link, did you mean differentiating between the different routers (like react-router vs next.js routing)?

> I wonder if it is better to differentiate the values that come from dotenv vs. not -ex. the context assigns the initial value from dotenv without asking the client to hydrate.

Good point! We could place dotenv vars under an env prop and have separate (e.g theming) or other props within the provider’s config object. I can address this in a follow up as we better define what else could sit in the config.

**Comment by sandrahoang686** [comment]
> But I think we should provide all route-triggering actions as callbacks within the components so that components don't have to handle routing directly (if that's feasible)
Yes, very much agree

File: app/scripts/context/config-context.tsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import React, { createContext } from 'react';
+
+interface VedauiConfig {
+  mapboxToken: string;
+  envApiStacEndpoint: string;
+  envApiRasterEndpoint: string;
```
Comment (sandrahoang686) [inline app/scripts/context/config-context.tsx:6]: ⛏️ : the endpoints are prefixed with `env` - maybe `envMapboxToken` to keep the standard 🤷🏼‍♀️

**Pull request merged**